ms/MSOper/MSMetaData.cc

```diff
@@ -3546,6 +3553,341 @@ void MSMetaData::_computeScanAndSubScanProperties(
     }
 }
 
+void MSMetaData::_computeScanAndSubScanPropertiesParallel(
+    SHARED_PTR<std::map<ScanKey, MSMetaData::ScanProperties> >& scanProps,
+    SHARED_PTR<std::map<SubScanKey, MSMetaData::SubScanProperties> >& subScanProps,
+    Bool showProgress
+) const {
+    uInt nrows = nRows();
+    SHARED_PTR<Vector<Int> > scans = _getScans();
+    SHARED_PTR<Vector<Int> > fields = _getFieldIDs();
+    SHARED_PTR<Vector<Int> > ddIDs = _getDataDescIDs();
+    SHARED_PTR<Vector<Int> > states = _getStateIDs();
+    SHARED_PTR<Vector<Double> > times = _getTimes();
+    SHARED_PTR<Vector<Int> > arrays = _getArrayIDs();
+    SHARED_PTR<Vector<Int> > observations = _getObservationIDs();
+    SHARED_PTR<Vector<Int> > ant1, ant2;
+    SHARED_PTR<QVD> exposureTimes = _getExposureTimes();
+    SHARED_PTR<QVD> intervalTimes = _getIntervals();
+    vector<uInt> ddIDToSpw = getDataDescIDToSpwMap();
+    _getAntennas(ant1, ant2);
+    uInt nchunks = min((uInt)1000, nrows);
+    uInt chunkSize = nrows/nchunks;
+    if (nrows % nchunks > 0) {
+        ++chunkSize;
+    }
+    SHARED_PTR<ProgressMeter> pm;
+    if (showProgress) {
+        LogIO log;
+        log << LogOrigin("MSMetaData", __func__, WHERE)
+            << LogIO::NORMAL << "Compute subscan properties " << LogIO::POST;
+        pm.reset(new ProgressMeter(0, nchunks, "Compute subscan info"));
+    }
+    pair<map<ScanKey, ScanProperties>, map<SubScanKey, SubScanProperties> > *fut =
+        new pair<map<ScanKey, ScanProperties>, map<SubScanKey, SubScanProperties> >[nchunks];
+    uInt doneChunks = 0;
+#pragma omp parallel for
+    for (uInt i=0; i<nchunks; ++i) {
+        fut[i] = _getChunkSubScanProperties(
+            scans, fields, ddIDs, states, times, arrays,
+            observations, ant1, ant2, exposureTimes,
+            intervalTimes,  ddIDToSpw,  i * chunkSize,
+            std::min((i + 1) * chunkSize - 1, nrows - 1)
+        );
+#pragma omp atomic
```

Comment:
now this can be moved into in the critical section, no need for the atomic anymore.